ddl/partition.go

```diff
@@ -216,3 +217,24 @@ func onDropTablePartition(t *meta.Meta, job *model.Job) (ver int64, _ error) {
 	job.Args = []interface{}{partitionID}
 	return ver, nil
 }
+
+func getPartitionIDs(table *model.TableInfo) []int64 {
+	if table.GetPartitionInfo() == nil {
+		return []int64{}
+	}
+	partitionIDs := make([]int64, 0, len(table.Partition.Definitions))
+	for _, def := range table.Partition.Definitions {
+		partitionIDs = append(partitionIDs, def.ID)
+	}
+	return partitionIDs
+}
+
+func genGlobalID(d *ddlCtx) (int64, error) {
```

Comment:
There is a function [here](https://github.com/pingcap/tidb/blob/master/ddl/ddl.go#L403). Can you merge them together?

Reply:
@zimulala @tiancaiamao Do you think you can merge?

Reply:
If you use `getGlobalID(store kv.Storage)`, they can actually be one.
You can still keep `func (d *ddl) genGlobalID() (int64, error)`, let it call your if there are still many places use it and you don't want irrelevant changes in this PR.

Reply:
You can solve this in another PR.